main/src/com/google/refine/commands/row/AddRowsCommand.java

```diff
@@ -58,37 +58,46 @@ public void doPost(HttpServletRequest request, HttpServletResponse response)
             return;
         }
 
+        processRequest(request, response);
+    }
+
+    private void processRequest(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
         try {
             Project project = getProject(request);
-            List<Row> rows = getRowData(request);
+            List<Row> rowList = getRowData(request);
             int insertionIndex = getInsertionIndex(request, project);
 
-            AbstractOperation op = new RowAdditionOperation(rows, insertionIndex);
-            Process process = op.createProcess(project, new Properties());
-
-            performProcessAndRespond(request, response, project, process);
+            performProjectOperation(request, response, project, rowList, insertionIndex);
         } catch (Exception e) {
             respondException(response, e);
         }
     }
 
+    private static void performProjectOperation(HttpServletRequest request, HttpServletResponse response, Project project, List<Row> rows,
+            int insertionIndex) throws Exception {
+        AbstractOperation op = new RowAdditionOperation(rows, insertionIndex);
+        Process process = op.createProcess(project, new Properties());
+
+        performProcessAndRespond(request, response, project, process);
+    }
+
     public int getInsertionIndex(HttpServletRequest request, Project project) {
-        String data = request.getParameter(INDEX_PARAMETER);
-        int index = Integer.parseInt(data);
+        String indexString = request.getParameter(INDEX_PARAMETER);
```

Comment:
While this is arguably a better variable name, the real issue with this code is that if the parameter is empty or badly formatted, `Integer.parseInt` is going to throw a NumberFormatException.